doc: README: document how to run the e2e suite (#182)
openshift-kni/numaresources-operator: README.md

* the NUMA-aware scheduling stack does not yet support alignment of [memory resources](https://kubernetes.io/docs/tasks/administer-cluster/memory-manager/)

Please check the issues section for the known issues and limitations of the NUMA resources operator.

## running the e2e suite against your cluster

The NUMA resources operator comes with a growing e2e suite to validate components of the stack (operator proper, RTE) as well as the NUMA aware scheduling as a whole.
Pre-built container images including the suites [are available](https://quay.io/repository/openshift-kni/numaresources-operator-tests?tab=info).
There is **no support** for these e2e tests images, and they are recommended to be used only for development/CI purposes.

The e2e suite can *optionally* setup/teardown the numaresources stack, but the suite expects a pre-configured kubeletconfig.
Please find [here](https://raw.githubusercontent.com/openshift-kni/numaresources-operator/main/doc/examples/kubeletconfig.yaml) an example of recommended kubeletconfig.

The e2e suite can set a default `kubeletconfig`, but this is not recommended. The recommended flow is to pre-configure the cluster with a `kubeletconfig` object.

> Should you decide to use the default `kubeletconfig`, please elide the `-e E2E_NROP_INSTALL_SKIP_KC=true` into all the `podman` command lines below.

Reviewer: ```suggestion
Should you decide to use the default `kubeletconfig`, please omit the `-e E2E_NROP_INSTALL_SKIP_KC=true` from all the `podman` command lines below.
```

I had to look up `elide` :)

* the NUMA-aware scheduling stack does not yet support alignment of [memory resources](https://kubernetes.io/docs/tasks/administer-cluster/memory-manager/)

Please check the issues section for the known issues and limitations of the NUMA resources operator.

## running the e2e suite against your cluster

The NUMA resources operator comes with a growing e2e suite to validate components of the stack (operator proper, RTE) as well as the NUMA aware scheduling as a whole.
Pre-built container images including the suites [are available](https://quay.io/repository/openshift-kni/numaresources-operator-tests?tab=info).
There is **no support** for these e2e tests images, and they are recommended to be used only for development/CI purposes.

> Before to run the e2e suite, please make sure the `kubeletconfig` is set up correctly.

Reviewer: ```suggestion
Before running the e2e suite, please make sure the `kubeletconfig` is set up correctly.
```

The e2e suite can *optionally* setup/teardown the numaresources stack, but the suite expects a pre-configured kubeletconfig.

> Pleae find [here](https://raw.githubusercontent.com/openshift-kni/numaresources-operator/main/doc/examples/kubeletconfig.yaml) an example of recommended kubeletconfig.

Reviewer: ```suggestion
Please find [here](https://raw.githubusercontent.com/openshift-kni/numaresources-operator/main/doc/examples/kubeletconfig.yaml) an example of recommended kubeletconfig.
```

and I would move the rows 21 and 22 above 20.

Reviewer: OR remove the line 20 completely

The e2e suite can set a default `kubeletconfig`, but this is not recommended. The recommended flow is to pre-configure the cluster with a `kubeletconfig` object.

> If you use the pre-configured cluster, you will need to add `-e E2E_NROP_INSTALL_SKIP_KC=true` to all the `podman` command lines below.

Reviewer: Just to check my understanding. The user is expected to apply KubeletConfig first and use this env var, right? Why not make it the default in the examples then and document the exceptional case as the exception instead.

Author: makes more sense this way, right

Reviewer: Especially since we should probably default to documenting the safe way (no changes to the cluster unless asked for).
